src/java.base/share/classes/java/lang/reflect/AccessFlag.java

```diff
@@ -0,0 +1,420 @@
+     * value of {@code 0x0020}.
+     * @see java.lang.module.ModuleDescriptor.Requires.Modifier#TRANSITIVE
+     */
+    TRANSITIVE(0x0000_0020, false, Set.of(Location.MODULE_REQUIRES)),
+
+    /**
+     * The access flag {@code ACC_SYNCHRONIZED}, corresponding to the
+     * source modifier {@link Modifier#SYNCHRONIZED synchronized} with
+     * a mask value of <code>{@value "0x%04x" Modifier#SYNCHRONIZED}</code>.
+     */
+    SYNCHRONIZED(Modifier.SYNCHRONIZED, true, Set.of(Location.METHOD)),
+
+    /**
+     * The module requires flag {@code ACC_STATIC_PHASE} with a mask
+     * value of {@code 0x0040}.
+     * @see java.lang.module.ModuleDescriptor.Requires.Modifier#STATIC
+     */
+    STATIC_PHASE(0x0000_0040, false, Set.of(Location.MODULE_REQUIRES)),
+
+     /**
+      * The access flag {@code ACC_VOLATILE}, corresponding to the
+      * source modifier {@link Modifier#VOLATILE volatile} with a mask
+      * value of <code>{@value "0x%04x" Modifier#VOLATILE}</code>.
+      */
+    VOLATILE(Modifier.VOLATILE, true, Set.of(Location.FIELD)),
+
+    /**
+     * The access flag {@code ACC_BRIDGE} with a mask value of
+     * <code>{@value "0x%04x" Modifier#BRIDGE}</code>
+     * @see Method#isBridge()
+     */
+    BRIDGE(Modifier.BRIDGE, false, Set.of(Location.METHOD)),
+
+    /**
+     * The access flag {@code ACC_TRANSIENT}, corresponding to the
+     * source modifier {@link Modifier#TRANSIENT transient} with a
+     * mask value of <code>{@value "0x%04x" Modifier#TRANSIENT}</code>.
+     */
+    TRANSIENT(Modifier.TRANSIENT, true, Set.of(Location.FIELD)),
+
+    /**
+     * The access flag {@code ACC_VARARGS} with a mask value of
+     <code>{@value "0x%04x" Modifier#VARARGS}</code>.
+     * @see Executable#isVarArgs()
+     */
+    VARARGS(Modifier.VARARGS, false, Set.of(Location.METHOD)),
+
+    /**
+     * The access flag {@code ACC_NATIVE}, corresponding to the source
+     * modifier {@link Modifier#NATIVE native} with a mask value of
+     * <code>{@value "0x%04x" Modifier#NATIVE}</code>.
+     */
+    NATIVE(Modifier.NATIVE, true, Set.of(Location.METHOD)),
+
+    /**
+     * The access flag {@code ACC_INTERFACE} with a mask value of
+     * {@code 0x0200}.
+     * @see Class#isInterface()
+     */
+    INTERFACE(Modifier.INTERFACE, false,
+              Set.of(Location.CLASS, Location.INNER_CLASS)),
+
+    /**
+     * The access flag {@code ACC_ABSTRACT}, corresponding to the
+     * source modifier {@link Modifier#ABSTRACT abstract} with a mask
+     * value of <code>{@value "0x%04x" Modifier#ABSTRACT}</code>.
+     */
+    ABSTRACT(Modifier.ABSTRACT, true,
+             Set.of(Location.CLASS, Location.METHOD, Location.INNER_CLASS)),
+
+    /**
+     * The access flag {@code ACC_STRICT}, corresponding to the source
+     * modifier {@link Modifier#STRICT strictfp} with a mask value of
+     * <code>{@value "0x%04x" Modifier#STRICT}</code>.
+     */
+    STRICT(Modifier.STRICT, true, Set.of(Location.METHOD)),
```

Comment:
ACC_STRICT is defined for class files and appears in the Class.getModifiers() before class file version 46. 
Also it is included in Modifer.classModifiers();  Modifer.CLASS_MODIFIERS.
it might be worth a note saying it is class file version specific.

Reply:
The ACC_STRICT bit is defined for class file major versions 46 through 60:
https://docs.oracle.com/javase/specs/jvms/se18/html/jvms-4.html
It is the first access flag to be retired in this way.

A number of the access flags (ENUM, ANNOTATION, VARARGS, BRIDGE) are only defined for JDK 5.0 and later, class file major version 49 and above. A similar situations hold for other flags introduced to support modularity only being defined for class file major versions 53 and above.

For JDK 20, the native class file major version is 64.

It would be possible to add some kind of ClassFileVersion enum or other API mechanisms to support a query like "what major versions are this access flag defined?" However, I don't think it is necessary to include such information in the core reflection API at this time and interested parties can lookup the information in the JVMS editions. (I think ClassFileVersion enum would be more appropriate for a byte code manipulation library.)

I'll work on at least adding an explanatory paragraph with a disclaimer that the defined access flags are class file version specific and this class is intended to capture class file versions over time.